zingo-testutils/src/chain_generic_tests.rs

```diff
@@ -128,6 +129,92 @@ pub mod fixtures {
         assert_eq!(expected_fee, recorded_fee);
     }
 
+    /// chain-generic shielding test with dust avoidance
+    /// only shield when it would return positive value
+    pub async fn shield_positive<CC>()
+    where
+        CC: ConductChain,
+    {
+        let mut environment = CC::setup().await;
+
+        let sender = environment.fund_client_orchard(1_000_000).await;
+        let shielder = environment.create_client().await;
+
+        // creates a bunch of transparent dust outputs for recipient
+        assert_eq!(
+            with_assertions::propose_send_bump_sync_recipient(
+                &mut environment,
+                &sender,
+                &shielder,
+                vec![
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000),
+                    (Transparent, 1_000)
+                ],
+            )
+            .await,
+            40_000
```

Comment:
can we atleast have these in multiples of marginal fee so the reviewer can count easier. I have said in many other PRs I dont think we should have magic number expected fees. we need an expected fee calculator that reviewed and used thourghout our testing

Reply:
addressed

Reply:
haha, 'magic number expected fees'

Reply:
good point, though there is also an internal assert that proposed = recorded

Reply:
fixed slightly